### PR TITLE
Use bitwise OR instead of addition

### PR DESCRIPTION
**Pull request by AtariDreams**

### CODE REVIEW

File: cups/http-support.c [hunk 1]
```diff
@@ -727,9 +727,9 @@ httpEncode64_2(char       *out,		/* I - String to write to */
     if (outptr < outend)
     {
       if (inlen > 1)
-        *outptr ++ = base64[(((in[0] & 255) << 4) | ((in[1] & 255) >> 4)) & 63];
+        *outptr ++ = base64[((in[0] << 4) | (in[1] >> 4)) & 63];
```
Comment (michaelrsweet) [inline cups/http-support.c:730]: "in" is sometimes signed char (implementation-specific), so we need to mask to 8 bits (sign extended when converted to int).

File: cups/ipp.c [hunk 2]
```diff
@@ -2910,9 +2910,8 @@ ippReadIO(void       *src,		/* I - Data source */
 
           ipp->request.any.version[0]  = buffer[0];
           ipp->request.any.version[1]  = buffer[1];
-          ipp->request.any.op_status   = (buffer[2] << 8) | buffer[3];
-          ipp->request.any.request_id  = (((((buffer[4] << 8) | buffer[5]) << 8) |
-	                        	 buffer[6]) << 8) | buffer[7];
+          ipp->request.any.op_status  = (buffer[2] << 8)  |  buffer[3];
```
Comment (michaelrsweet) [inline cups/ipp.c:2913]: Fix whitespace

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
See comments.

File: cups/http-support.c (hunk 1 above)
Comment (AtariDreams) [reply cups/http-support.c:730]: Bitwise operations make that redundant. Like "& 63" already does the job and so does the << 4

Comment (michaelrsweet) [reply cups/http-support.c:730]: Without masking, the interior `in[1] >> 4` will get sign-extended so the upper 4 bits will be 1's. If "in" was `unsigned char` then it wouldn't be an issue, but "char" can be signed or unsigned.

Comment (AtariDreams) [reply cups/http-support.c:730]: & 63 already does the necessary masking is what I am trying to say

Comment (michaelrsweet) [reply cups/http-support.c]: No, it doesn't.

`-1 >> 4` is still -1 (all 1's), which will mask the bits from the first part. `(-1 & 255) >> 4` is 15 (255 >> 4). So we need to mask `in[1]`...

Comment (AtariDreams) [reply cups/http-support.c]: Okay, addressed

**Comment by AtariDreams** [comment]
@michaelrsweet Addressed!

**Review by michaelrsweet — APPROVED** [review]

**Pull request merged**